### DISCUSSION TITLE
Inquiry on Retrieving Transport Order Status and Allocated Resources Changes in Custom OpenTCS Adapter

### DISCUSSION THREAD

**Discussion by BenYaho** [openTCS/opentcs#129] [Q&A] [answered]
Dear OpenTCS Support Team,

I am currently developing a custom adapter in OpenTCS 5.7.0 and would like to inquire about the following functionalities:

How can I retrieve the status of a transport order to check if it is still active or if there have been any changes in its status?
Is there a way to monitor changes in the AllocatedResources of a vehicle? I am particularly interested in receiving events when such changes occur.
I appreciate your assistance in providing guidance or documentation on implementing these features in a custom OpenTCS adapter.

Thank you for your time and support.

**Comment by swltr** [comment, marked as answer]
@BenYaho openTCS 5.7 is relatively old, and changes related to your question have been made in the meantime. I recommend updating to 5.12.

About checking whether a transport order is still active: What would you need that for? If the transport order had been withdrawn, the vehicle driver wouldn't receive any further movement commands. So, as long as you receive movement commands, the transport order has not been withdrawn. It might help to understand your use case better here.

As for monitoring changes of allocated resources: Allocated resources are reflected in the `Vehicle` object, and you could use `TCSObjectEvent`s to be notified of changes. (Not in 5.7, though.)
